src/toaster/ToastContext.ts

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export interface ToastContextProps {
+  usedToaster?: boolean;
+}
+
+const ToastContext = React.createContext<ToastContextProps>({ usedToaster: false });
```

Comment:
The default value of `ToastContext` could be `undefined`, and child component could simply check whether its within a `ToasterContext`. That way we don't have to add this `usedToaster` property which is always set `true` whenever `ToastContext.Provider` is used.

```js
const toastContext = useContext(ToastContext);
const usedToaster = Boolean(toastContext)
```

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L328
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L377

In React, setting a defaultValue of `undefined` for Context is deprecated. The value prop of Context.Provider is also a required prop.

Reply:
Passing `undefined` does not mean considering `defaultValue` optional, but to consider `ToastContext` nullable.

```js
const ToastContext = React.createContext<ToastContextProps | undefined>(undefined);

// or use null
const ToastContext = React.createContext<ToastContextProps | null>(null);
```

The point is that a "usedToaster" looks so ambiguous as a property of a "ToastContext". When you _call `useContext(ToastContext)` and get `usedToaster: false`_, it sounds like "yes, but no".

Reply:
This would have to set a meaningless value for Context.Provider since its value property is required.
```tsx
<Context.Provider value={null}>
```

I think it is also possible without Context. Add a `usedToaster` prop to `<Message>` that is not exposed to users.

Reply:
> This would have to set a meaningless value for Context.Provider since its value property is required.
> 
> ```tsx
> <Context.Provider value={null}>
> ```

Not really. The value passed to `ToastContext.Provider` will not be `null` but info about the toast (defined as `ToastContextProps`) - currently none though.

```jsx
<ToastContext.Provider value={{ placement, duration, containerId, ...thingsLikeThat }}>
</ToastContext.Provider>
```
 
> Add a `usedToaster` prop to `<Message>` that is not exposed to users.

I don't think it's a good idea to add extra props around the closing behavior to Message, while we're about to remove the closing logic from it in the future.

---

To conclude, I believe current implementation is workable. I'm going to merge it as is.